### PR TITLE
Update if_changed.sh

### PR DESCRIPTION
**Pull request by Nicofuma**

### CODE REVIEW

File: if_changed.sh [hunk 1]
```diff
@@ -1,10 +1,16 @@
+$ cat test.sh
```
Comment (callumacrae) [inline if_changed.sh]: wat

File: if_changed.sh [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
 
+PWD=$(pwd)
+
 changed_files=$(git diff-tree -r --name-only ORIG_HEAD HEAD)
 
 function if_changed() {
-  [[ "$changed_files" == *"$1"* ]] && eval "$2"
+  files=$(printf "%s" "$changed_files" | grep "$1$" -)
+  for file in $files; do
+    cd $(dirname $file) && eval "$2" && cd $PWD
```
Comment (callumacrae) [inline if_changed.sh]: I think you should be able to use `cd -` here to return to the previous directory. Might break if the eval contains a cd call, though

**Comment by callumacrae** [comment]
Thanks! I've added you as a collaborator because you seem to know more bash than I do.

edit: note to self for tomorrow, test this by having a file in a subdirectory in the test branch and running `pwd`

File: if_changed.sh (hunk 1 above)
Comment (Nicofuma) [inline if_changed.sh]: oups

**Pull request merged**